File: examples/python/openai_chat.py
"""
BOB Gemini Free - OpenAI Python SDK Example
Run BOB Gemini Free gateway: ./bob-gemini-free --port 8081
Then run: pip install openai && python openai_chat.py
"""

from openai import OpenAI

# 1. Initialize client pointing to local BOB gateway
client = OpenAI(
    base_url="http://127.0.0.1:8081/v1",
    api_key="none",  # or your configured BOB_GEMINI_FREE_API_KEYS
)

print("--- 1. Synchronous Completion ---")
response = client.chat.completions.create(
    model="gemini-3.7-flash",
    messages=[
        {"role": "user", "content": "Explain quantum superposition in 2 sentences."}
    ],
)
print("Response:\n", response.choices[0].message.content)

print("\n--- 2. Real-Time Streaming with Deep Reasoning ---")
stream = client.chat.completions.create(
    model="gemini-3.7-flash-thinking",
    messages=[
        {"role": "user", "content": "Solve: What is the sum of the first 10 prime numbers?"}
    ],
    stream=True,
)

for chunk in stream:
    delta = chunk.choices[0].delta if chunk.choices else None
    if delta:
        # Check for real-time reasoning content
        if getattr(delta, "reasoning_content", None):
            print(f"[Thinking]: {delta.reasoning_content}", end="", flush=True)
        # Check for clean answer content
        if delta.content:
            print(delta.content, end="", flush=True)

print("\n\nDone!")

File: examples/nodejs/openai_chat.mjs
/**
 * BOB Gemini Free - Node.js OpenAI SDK Example
 * Run BOB Gemini Free gateway: ./bob-gemini-free --port 9610
 * Then run: npm install openai && node openai_chat.mjs
 */

import OpenAI from 'openai';

const openai = new OpenAI({
  baseURL: 'http://127.0.0.1:9610/v1',
  apiKey: 'none',
});

async function main() {
  console.log('--- 1. Synchronous Chat Completion ---');
  const completion = await openai.chat.completions.create({
    model: 'gemini-3.7-flash',
    messages: [{ role: 'user', content: 'Explain event loops in Node.js in 2 sentences.' }],
  });
  console.log(completion.choices[0].message.content);

  console.log('\n--- 2. Real-Time Streaming ---');
  const stream = await openai.chat.completions.create({
    model: 'gemini-3.7-flash-thinking',
    messages: [{ role: 'user', content: 'Count from 1 to 5 with a short explanation for each.' }],
    stream: true,
  });

  for await (const chunk of stream) {
    const delta = chunk.choices[0]?.delta;
    if (delta?.reasoning_content) {
      process.stdout.write(`[Thinking]: ${delta.reasoning_content}`);
    }
    if (delta?.content) {
      process.stdout.write(delta.content);
    }
  }
  console.log('\n\nDone!');
}

main().catch(console.error);
